datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -293,14 +313,15 @@ func (f *ChainDataFetcher) resLoop() {
 			logger.Info("stopped resLoop for chaindatafetcher")
 			return
 		case res := <-f.resCh:
-			if res.err != nil {
-				logger.Error("db insertion is failed", "blockNumber", res.blockNumber, "reqType", res.reqType, "err", res.err)
-				// TODO-ChainDataFetcher add retry logic when data insertion is failed
-			} else {
-				if err := f.updateCheckpoint(res.blockNumber.Int64()); err != nil {
-					logger.Error("Failed to update checkpoint", "err", err, "checkpoint", res.blockNumber.Int64())
+			for {
+				err := f.updateCheckpoint(res.blockNumber.Int64())
+				if err != nil {
+					logger.Error("Failed to update checkpoint. Sleeping for retry...", "err", err, "checkpoint", res.blockNumber.Int64(), "interval", DBInsertRetryInterval)
+					time.Sleep(DBInsertRetryInterval)
+				} else {
+					logger.Info("responsed", "blockNumber", res.blockNumber.Uint64())
+					break
```

Comment:
Can we merge these lines into `for` loop?

Reply:
I removed the retry logic for checkpoint https://github.com/klaytn/klaytn/pull/610#discussion_r469024987.